src/components/forms/TextInput/TextInput.tsx

```diff
@@ -28,43 +28,51 @@ export type OptionalTextInputProps = CustomTextInputProps &
 
 export type TextInputProps = RequiredTextInputProps & OptionalTextInputProps
 
-export const TextInput = ({
-  id,
-  name,
-  type,
-  className,
-  validationStatus,
-  inputSize,
-  inputRef,
-  ...inputProps
-}: TextInputProps): React.ReactElement => {
-  const isError = validationStatus === 'error'
-  const isSuccess = validationStatus === 'success'
-  const isSmall = inputSize === 'small'
-  const isMedium = inputSize === 'medium'
+export const TextInput = forwardRef(
+  (
+    props: TextInputProps,
+    ref: React.ForwardedRef<HTMLInputElement> | undefined
+  ): React.ReactElement => {
+    const {
+      id,
+      name,
+      type,
+      className,
+      validationStatus,
+      inputSize,
+      inputRef,
+      ...inputProps
+    } = props
 
-  const classes = classnames(
-    'usa-input',
-    {
-      'usa-input--error': isError,
-      'usa-input--success': isSuccess,
-      'usa-input--small': isSmall,
-      'usa-input--medium': isMedium,
-    },
-    className
-  )
+    const isError = validationStatus === 'error'
+    const isSuccess = validationStatus === 'success'
+    const isSmall = inputSize === 'small'
+    const isMedium = inputSize === 'medium'
 
-  return (
-    <input
-      data-testid="textInput"
-      className={classes}
-      id={id}
-      name={name}
-      type={type}
-      ref={inputRef}
-      {...inputProps}
-    />
-  )
-}
+    const classes = classnames(
+      'usa-input',
+      {
+        'usa-input--error': isError,
+        'usa-input--success': isSuccess,
+        'usa-input--small': isSmall,
+        'usa-input--medium': isMedium,
+      },
+      className
+    )
+
+    return (
+      <input
+        data-testid="textInput"
+        className={classes}
+        id={id}
+        name={name}
+        type={type}
+        ref={inputRef || ref}
+        {...inputProps}
+      />
+    )
+  }
+)
 
+TextInput.displayName = 'TextInput'
```

Comment:
A bit weird, but the existing pattern for making sure the components have a display name is done like so:
- https://github.com/trussworks/react-uswds/blob/4513e51c04caa6c6f83ae57281b8419d874ab894/src/components/forms/DatePicker/Day.tsx#L27
- https://github.com/trussworks/react-uswds/blob/4513e51c04caa6c6f83ae57281b8419d874ab894/src/components/forms/DatePicker/Day.tsx#L113

I don't remember if there was a reason we did it that way, or if your approach is better, so I won't block on it

Reply:
I am happy to change but I find it a bit strange - this approach makes it so that the display name of the component (shows up in dev tools, error messages) will be `DayForwardRef` not `Day` which I find somewhat unideal.

Reply:
I agree with you. Let's leave it as is. Can always update easily if we find any problems or want to unify one way or the other